torch/_inductor/memory.py

```diff
@@ -28,48 +29,35 @@ class MemoryPlanningInfoForBuffer:
     succ_nodes: OrderedSet[BaseSchedulerNode] = dataclasses.field(
         default_factory=OrderedSet
     )
-    outdegree: int = 0  # this is used only in topological_sort_lpmf
```

Comment:
In `MemoryPlanningInfoForBuffer`, I am only keeping the attributes that are static. For dynamically set attributes (e.g., `outdegree`), will keep track of them during the function that needs them. 

Similarly, removed e.g., `memory_to_free` and `indegree` from `MemoryPlanningInfoForNode`.